nightfall-optimist/src/event-handlers/subscribe.mjs

```diff
@@ -96,10 +96,7 @@ export async function startEventQueue(lastSyncedBlock, callback, ...arg) {
   ];
   const contracts = await Promise.all(contractNames.map(c => waitForContract(c)));
   const emitters = contracts.map(e => {
-    let emitterC;
-    // if we've just resynced, start from the end of the sync:
-    if (lastSyncedBlock) emitterC = e.events.allEvents({ fromBlock: lastSyncedBlock + 1 });
-    else emitterC = e.events.allEvents();
+    const emitterC = e.events.allEvents();
```

Comment:
Every time optimist is restarted but the database is intact, it starts syncing from the start of the contract ?
This would be inefficient if syncing after a few years

Reply:
No, that won't happen.  Without any argument, `.allEvents` will start listening for events from the current block onwards.  We're just telling it to subscribe to all events that come along.